awx/api/validators.py

```diff
@@ -0,0 +1,69 @@
+import re
+
+from django.core.validators import RegexValidator, MaxLengthValidator, validate_ipv46_address
+from django.core.exceptions import ValidationError
+
+
+class HostnameRegexValidator(RegexValidator):
+    """
+    Fully validates a domain name that is compliant with norms in Linux/RHEL
+        - Cannot start with a hyphen
+        - Cannot begin with, or end with a "."
+        - Cannot contain any whitespaces
+        - Entire hostname is max 255 chars (including dots)
+        - Each domain/label is between 1 and 63 characters, except top level domain, which must be at least 2 characters
+        - Supports ipv4, ipv6, simple hostnames and FQDNs
+        - Follows RFC 9210 (modern RFC 1123, 1178) requirements
+
+    Accepts an IP Address or Hostname as the argument
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        if 'regex' not in kwargs.keys():
+            self.regex = '^[a-z0-9][-a-z0-9]*$|^([a-z0-9][-a-z0-9]{0,62}[.])*[a-z0-9][-a-z0-9]{1,62}$'
+        if 'flags' not in kwargs.keys():
+            self.flags = re.IGNORECASE
+
+        super().__init__(**kwargs)
+
+    def __call__(self, value):
+        regex_matches, err = self.__validate(value)
+        invalid_input = regex_matches if self.inverse_match else not regex_matches
+        if invalid_input:
+            if err is None:
+                err = ValidationError(self.message, code=self.code, params={"value": value})
+            raise err
+
+    def __str__(self):
+        return f"regex={self.regex}, message={self.message}, code={self.code}, inverse_match={self.inverse_match}, flags={self.flags}"
+
+    def __validate(self, value):
+
+        # import sdb; sdb.set_trace();
+
+        if len(value.split(" ")) > 1:
```

Comment:
if we are going to continue to make this flexible (which I think we shouldn't -- the semantics of valid hostnames aren't going to change between usages), this can be expressed more elegantly as `if ' ' in value:`.